src/Frame.cc

```diff
@@ -758,18 +758,18 @@ bool Frame::CalculateHistogram(int region_id, int channel, int stokes, int num_b
         }
         bool write_lock(false);
         tbb::queuing_rw_mutex::scoped_lock cache_lock(_cache_mutex, write_lock);
-        CalcHistogram(num_bins, stats, _image_cache, results);
+        hist = CalcHistogram(num_bins, stats, _image_cache);
```

Comment:
I don't like this copy. CalcHistogram returns a histogram, that could actually be _moved_...

Reply:
Won't it be implicitly moved?

Reply:
Yes, good point, it should be implicitly moved; I think overthought that `=`

Reply:
Or pass hist by reference to CalcHistogram, which fills the values.

Reply:
or pass hist by reference

Reply:
I think it's the same behavior. Either at this point or inside CalcHistogram the histogram is constructed and placed into hist.